src/ae/contract.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Contract module - routines to interact with the æternity contract
+ *
+ * High level documentation of the contracts are available at
+ * https://github.com/aeternity/protocol/tree/master/contracts and
+ *
+ * @module @aeternity/aepp-sdk/es/ae/contract
+ * @export Contract
+ * @example import { Contract } from '@aeternity/aepp-sdk'
+ */
+
+// @ts-expect-error TODO remove
+import { decode, produceNameId } from '../tx/builder/helpers'
+// @ts-expect-error TODO remove
+import { getNetworkId } from '../node'
+import { EncodedData, EncodingType } from '../utils/encoder'
+import getContractIns from '../contract/aci'
+
+// TODO remove and import from node once it's merged
+type GetNetworkIdOptions = any
+type SignOptions = object
+type DelegateSigOptions = GetNetworkIdOptions & SignOptions
+
+/**
+ * Utility method to create a delegate signature for a contract
+ * @alias module:@aeternity/aepp-sdk/es/ae/contract
+ * @category async
+ * @param ids The list of id's to prepend
+ * @param opt Options
+ * @return Signature in hex representation
+ */
+async function delegateSignatureCommon (
+  onAccount: any,
+  ids: Array<EncodedData<EncodingType>> = [],
+  opt: DelegateSigOptions = {}): Promise<string> {
```

Comment:
```suggestion
  opt: DelegateSigOptions = {}
): Promise<string> {
```
just easier to read, also it would be consistent with other code in this repo